dashboard/test/controllers/admin_users_controller_test.rb

```diff
@@ -257,27 +260,121 @@ class AdminUsersControllerTest < ActionController::TestCase
+    assert_equal 0, @user.user_levels_by_level(@script).count
+  end
+
+  test "delete_progress deletes channel tokens" do
+    # TODO: Write this test after the work to add script_ids to channel tokens is completed.
+  end
+
+  test "delete_progress deletes teacher feedback" do
+    sign_in @admin
+
+    teacher = create(:teacher)
+    student = create(:student)
+    section = create(:section, teacher: teacher)
+    section.add_student(student)
+
+    TeacherFeedback.create!(teacher: teacher, student: student, script: @script, level: @level1)
+    TeacherFeedback.create!(teacher: teacher, student: student, script: @script, level: @level1)
+    TeacherFeedback.create!(teacher: teacher, student: student, script: @script, level: @level2)
+    TeacherFeedback.create!(teacher: teacher, student: student, script: @script, level: @level3)
+    assert_equal 3, TeacherFeedback.get_latest_feedbacks_received(student.id, nil, @script.id).count
+
+    post :delete_progress, params: {user_id: student.id, script_id: @script.id, reason: 'Testing'}
+    assert_equal 0, TeacherFeedback.get_latest_feedbacks_received(student.id, nil, @script.id).count
+  end
+
+  test "delete_progress for driver leaves pairing record" do
+    driver = create(:student)
+    navigator = create(:student)
+
+    driver_user_level = UserLevel.create!(user: driver, script: @script, level: @level, best_result: 100)
+    navigator_user_level = UserLevel.create!(user: navigator, script: @script, level: @level, best_result: 100)
+    PairedUserLevel.create!(driver_user_level_id: driver_user_level.id, navigator_user_level_id: navigator_user_level.id)
+
+    # check that we've correctly setup the pairing relationship
+    assert navigator_user_level.navigator?
+    assert_equal 1, PairedUserLevel.pairs(navigator_user_level).count
+
+    # delete the driver's progress
+    post :delete_progress, params: {user_id: driver.id, script_id: @script.id, reason: 'Testing'}
+
+    # navigator should still be able to see that they were paired on this level
+    # (but can no longer tell who they were paired with)
+    navigator_user_level.reload
+    assert navigator_user_level.navigator?
+    assert_equal 1, PairedUserLevel.pairs(navigator_user_level).count
+  end
+
+  test "delete_progress for navigator leaves pairing record" do
+    driver = create(:student)
+    navigator = create(:student)
+
+    driver_user_level = UserLevel.create!(user: driver, script: @script, level: @level, best_result: 100)
+    navigator_user_level = UserLevel.create!(user: navigator, script: @script, level: @level, best_result: 100)
+    PairedUserLevel.create!(driver_user_level_id: driver_user_level.id, navigator_user_level_id: navigator_user_level.id)
+
+    # check that we've correctly setup the pairing relationship
+    assert driver_user_level.driver?
+    assert_equal 1, PairedUserLevel.pairs(driver_user_level).count
+
+    # delete the driver's progress
```

Comment:
Should be `# delete the navigator's progress`

Reply:
Good catch, thanks!